src/woo_publications/publications/tests/test_publications_api.py

```diff
@@ -1891,3 +1891,120 @@ def test_updated_publication_schedules_index_task(
                 mock_index_publication_delay.assert_called_once_with(
                     publication_id=publication.pk
                 )
+
+
+class PublicationApiRequiredFieldsTestCase(TokenAuthMixin, APITestCase):
+    def test_create_publication_status_concept(self):
+        url = reverse("api:publication-list")
+
+        data = {
+            "publicatiestatus": PublicationStatusOptions.concept,
+        }
+
+        response = self.client.post(url, data, headers=AUDIT_HEADERS)
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(
+            response.json(), {"officieleTitel": [_("This field is required.")]}
+        )
+
+    def test_create_publication_status_publish(self):
+        url = reverse("api:publication-list")
+
+        data = {
+            "publicatiestatus": PublicationStatusOptions.published,
+        }
+
+        response = self.client.post(url, data, headers=AUDIT_HEADERS)
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(
+            response.json(),
+            {
+                "officieleTitel": [_("This field is required.")],
+                "informatieCategorieen": [_("This field is required.")],
+                "publisher": [_("This field is required.")],
+            },
+        )
+
+    def test_update_publication_status_concept(self):
+        publication = PublicationFactory.create(
+            publisher=None,
+            publicatiestatus=PublicationStatusOptions.concept,
+        )
+        url = reverse(
+            "api:publication-detail",
+            kwargs={"uuid": str(publication.uuid)},
+        )
+
+        data = {
+            "publicatiestatus": PublicationStatusOptions.concept,
+            "officieleTitel": "",
+        }
+
+        response = self.client.put(url, data, headers=AUDIT_HEADERS)
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(
+            response.json(),
+            {
+                "officieleTitel": [_("This field cannot be null.")],
```

Comment:
`""` becomes `null` somewhere? Why?

Reply:
This is a default drf validator and this is basically saying that the field isn't allowed to be empty.
Which is different if the field isn't supplied at all which is when it will say that it is required.

Because the data is already filled out in the database this is what we need to do to test the updating of the data.

Reply:
shouldn't it say `blank` instead of `null`? https://github.com/encode/django-rest-framework/blob/e454758fb6edf1dcf5aa5417a388b940c871469c/rest_framework/fields.py#L725

Reply:
The dutch translation is `Dit veld mag niet leeg zijn.` so I am sure that it's fine.